packages/core/src/report/generate-report.ts

```diff
@@ -107,20 +127,19 @@ export async function generateReport(cwd: string): Promise<Report> {
       }),
   };
 
-  report.suggestions = [
-    await packagesWithPinnedVersions({ arboristValues }),
+  const suggestionInput = { arboristValues, rootArboristNode };
 
-    await packagesWithExtraArtifacts({ arboristValues }),
+  let suggestionsFromPlugins: Suggestion[] = [];
 
-    await notBeingAbsorbedByTopLevel({ rootArboristNode, arboristValues }),
+  for (const plugin of processedPlugins) {
+    if (plugin.getSuggestions) {
+      const suggestions = await plugin?.getSuggestions(suggestionInput);
 
-    await nestedDependencyFreshness({ rootArboristNode, arboristValues }),
+      suggestionsFromPlugins = suggestionsFromPlugins.concat(suggestions);
+    }
+  }
 
-    await topLevelDepsFreshness({
-      rootArboristNode,
-      arboristValues,
-    }),
-  ];
+  report.suggestions = [...suggestionsFromPlugins];
```

Comment:
Couple of things:

1. Every suggestion would need to have a `pluginID` added to its data
2. This is no longer an Array rather its a `{ [pluginID: string]: Suggestion[] }`

Reply:
Good call, I will add this.